clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -526,9 +527,12 @@ private void handleShareAcknowledgeSuccess(Node fetchTarget,
 
             if (!handler.handleResponse(response, requestVersion)) {
                 acknowledgeRequestState.onFailedAttempt(currentTimeMs);
-                if (response.error() != Errors.INVALID_SHARE_SESSION_EPOCH) {
+                if (response.error().exception() instanceof RetriableException || response.error() != Errors.INVALID_SHARE_SESSION_EPOCH) {
```

Comment:
This condition doesn't make sense to me. What about unrecoverable errors? I think they'll pass the test when they should not.

Reply:
Yeah right, my bad, have fixed the condition now.